core/build.gradle.kts

```
@@ -1,12 +1,13 @@
 import kotlinx.team.infra.mavenPublicationsPom
+import org.jetbrains.kotlin.gradle.targets.js.dsl.ExperimentalWasmDsl
 
 plugins {
     id("kotlin-multiplatform")
     `maven-publish`
 }
 
 base {
-    archivesBaseName = "kotlinx-collections-immutable" // doesn't work
+    archivesName.set("kotlinx-collections-immutable") // doesn't work
```

```suggestion
    archivesBaseName = "kotlinx-collections-immutable" // doesn't work
```

This assignment could be used on newer Gradle.

https://blog.gradle.org/simpler-kotlin-dsl-property-assignment